Task1: Implemented add user Api

neemesh-707 opened the pull request.

neemesh-707 committed:
Created a user entity class

neemesh-707 committed:
Implemented Api add user

AsmaSolver commented on src/main/java/com/solv/inventory/controller/UserController.java:
```diff
@@ -17,10 +15,9 @@
-
     @ApiOperation("- register new user")
-    @PostMapping("/")
-    public String newUser(@RequestBody String userName) {
```
Dont use entity object here use DTO object to send request to service

neemesh-707 committed:
Task1:Changes after Review:
Implemented DTO to send request to the service, added validation for different entity fields and added response class to show appropriate messages in case validation fails.
->Created API for updating a user ,getting all users, getting user by id,and updating the user.
->Added Global Exception class to handle exceptions.

neemesh-707 committed:
Removed unused code

neemesh-707 committed:
Task1:Changes after Review:
Implemented DTO to send request to the service, added validation for different entity fields and added response class to show appropriate messages in case validation fails.
->Created API for updating a user ,getting all users, getting user by id,and updating the user.
->Added Global Exception class to handle exceptions.
->Modified service layer of user api's to give common format output in case of failure and success
->Implemented test cases for service layer for user api's
->Modified HTTP status codes to give proper codes in case of success and failure.

neemesh-707 committed:
Task1:Modified Item controller to give same response in case of success and failure
